src/basic/cgroup-util.c

```diff
@@ -1688,6 +1688,25 @@ int cg_slice_to_path(const char *unit, char **ret) {
         return 0;
 }
 
+int cg_is_threaded(const char *controller, const char *path) {
+        _cleanup_free_ char *fs = NULL, *contents = NULL;
+        int r;
+
+        r = cg_get_path(controller, path, "cgroup.type", &fs);
+        if (r < 0)
+                return r;
+
+        r = read_full_virtual_file(fs, &contents, NULL);
+        if (r == -ENOENT)
+                return false; /* Assume no. */
+        if (r < 0)
+                return r;
+
+        /* If the cgroup is in the threaded mode, it contains "threaded".
+         * If one of the parents is in the threaded more, it may contain "invalid". */
+        return !!STRSTR_IN_SET(contents, "threaded", "invalid");
```

Comment:
hmm, strstr() is icky. I mean, one day kernel people might add a new value "non-threaded" or so, and strstr("threaded") will still consider it threaded...

maybe use strv_split() and then strv_contains()? i.e. make sure to use whitespace as field separators, so that we aren't confused by the above